sound/soc/sof/intel/hda-dai.c

```diff
@@ -454,6 +454,48 @@ static const struct snd_soc_dai_ops ipc3_hda_dai_ops = {
 	.prepare = ipc3_hda_dai_prepare,
 };
 
+static int hda_dai_suspend(struct hdac_bus *bus)
+{
+	struct snd_soc_pcm_runtime *rtd;
+	struct hdac_ext_stream *hext_stream;
+	struct hdac_ext_link *link;
+	struct hdac_stream *s;
+	const char *name;
+	int stream_tag;
+
+	/* set internal flag for BE */
+	list_for_each_entry(s, &bus->stream_list, list) {
+		hext_stream = stream_to_hdac_ext_stream(s);
+
+		if (!hext_stream)
+			return -EINVAL;
```

Comment:
Can this ever happen? Should we skip to the next stream in case it does instead braking out and return?

Reply:
not, it can't happen. I'll remove this, thanks for the comment @ujfalusi